controller/handler/trial.js

```diff
@@ -21,7 +21,17 @@ const httpNotFound = 404;
 function trialView (request, reply) {
     const trial = database.sequelize.model('trial');
     const stage = database.sequelize.model('stage');
-    const startDate = moment().startOf('Week');
+    const fromDate = request.query.fromDate;
+    const toDate = request.query.toDate;
+    const addSevenDays = 7;
+    let fromDatestr = null;
+    let toDatestr = null;
+
+    if ((typeof fromDate !== 'undefined' || fromDate !== null) && (typeof toDate !== 'undefined'
+   || toDate !== null)) {
+        fromDatestr = fromDate.toISOString();
+        toDatestr = toDate.toISOString();
+    }
```

Comment:
This guard statement is not needed.
`toISOString` should be called in the actual database replacement.